third_party/move/evm/move-to-yul/src/functions.rs

```diff
@@ -452,9 +452,9 @@ impl<'a> FunctionGenerator<'a> {
                         print_loc();
                         self.write_ref(
                             ctx,
-                            &get_local_type(srcs[1]),
-                            local(&srcs[0]),
+                            &get_local_type(srcs[0]),
                             local(&srcs[1]),
```

Comment:
Swapping this as part of this PR is problematic. What is about the prover and the endless other code usages of this, also outside of our code (e.g. decompiler)? Also isn't it the case that we have `WriteRef dest_ptr, src` in the bytecode, which is logically more valid than `WriteRef src, dest_ptr`, whatever file format does here?

Reply:
I believe I have changed all the instances of the code usages within aptos-core (not the decompiler, ofc).

This is the only stackless bytecode instruction where the arguments are required in the opposite order compared to file format bytecode, and there was a TODO as well to fix this. I guess the file format bytecode cannot be changed, so the alternative was changing the stackless bytecode.

Reply:
The alternative is to just keep as is and remove the TODO. Only because WriteRef has a weird stack order in file format, we do not need to adapt the same in stackless bytecode. 

I'm sure if we change this any external user of stackless bytecode will go through a painful experience, because this is not caught by the type system.